cpu/rpx0xx/periph/usbdev.c

```diff
@@ -0,0 +1,766 @@
+}
+
+/* available: is the processor in control of the buffer? */
+static bool _ep_buf_is_available(usbdev_ep_t *ep)
+{
+    __IOM uint32_t buf_ctrl_reg = *_get_ep_buf_ctrl_reg(ep->num, ep->dir);
+    return !(buf_ctrl_reg & USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_AVAILABLE_0_Msk);
+}
+
+/*
+static bool _ep_buf_is_full(usbdev_ep_t *ep)
+{
+    __IOM uint32_t buf_ctrl_reg = *_get_ep_buf_ctrl_reg(ep->num, ep->dir);
+    return buf_ctrl_reg & USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_FULL_0_Msk;
+}
+*/
+
+static size_t _ep_get_available(usbdev_ep_t *ep)
+{
+    if (_ep_buf_is_available(ep)) {
+        if (ep->dir == USB_EP_DIR_IN) {
+            rpx0xx_usb_ep_t *hw_ep = _get_ep(ep->num, ep->dir);
+            return hw_ep->dpram_bufsize;
+        } else {
+            __IOM uint32_t *reg = _get_ep_buf_ctrl_reg(ep->num, ep->dir);
+            return *reg | USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_LENGTH_0_Msk;
+        }
+    }
+    return 0UL;
+}
+
+static int _usbdev_ep_get(usbdev_ep_t *ep, usbopt_ep_t opt,
+                          void *value, size_t max_len)
+{
+    (void)max_len;
+    int ret = -ENOTSUP;
+
+    switch (opt) {
+    case USBOPT_EP_STALL:
+        assert(max_len == sizeof(usbopt_enable_t));
+        *(usbopt_enable_t *)value = _ep_get_stall(ep);
+        ret = sizeof(usbopt_enable_t);
+        break;
+    case USBOPT_EP_AVAILABLE:
+        assert(max_len == sizeof(size_t));
+        *(size_t *)value = _ep_get_available(ep);
+        ret = sizeof(size_t);
+        break;
+    default:
+        DEBUG("[rpx0xx usb]: Unhandled get call: 0x%x\n", opt);
+        break;
+    }
+    return ret;
+}
+
+static int _usbdev_ep_set(usbdev_ep_t *ep, usbopt_ep_t opt,
+                          const void *value, size_t value_len)
+{
+    (void)value_len;
+    int ret = -ENOTSUP;
+
+    switch (opt) {
+    case USBOPT_EP_ENABLE:
+        assert(value_len == sizeof(usbopt_enable_t));
+        if (*((usbopt_enable_t *)value)) {
+            _ep_enable(ep);
+        }
+        else {
+            _ep_disable(ep);
+        }
+        ret = sizeof(usbopt_enable_t);
+        break;
+    case USBOPT_EP_STALL:
+        assert(value_len == sizeof(usbopt_enable_t));
+        _ep_set_stall(ep, *(usbopt_enable_t *)value);
+        ret = sizeof(usbopt_enable_t);
+        break;
+    default:
+        DEBUG("[rpx0xx usb]: Unhandled set call: 0x%x\n", opt);
+        break;
+    }
+    return ret;
+}
+
+static void _mark_ep_buf_ready(usbdev_ep_t *ep)
+{
+    uint32_t buf_filled_bit = 0UL;
+    if (ep->dir == USB_EP_DIR_IN) {
+        buf_filled_bit = USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_FULL_0_Msk;
+    }
+    _ep_buf_ctrl_reg_write(ep->num, ep->dir, 
+                           USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_FULL_0_Pos, 
+                           buf_filled_bit);
+    _ep_buf_ctrl_reg_write(ep->num, ep->dir, 
+                           USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_AVAILABLE_0_Pos, 
+                           USBCTRL_DPRAM_EP0_IN_BUFFER_CONTROL_AVAILABLE_0_Msk);
```

Comment:
It seems to me that you're also missing to set (at least) the length field in the Buffer Control register of the endpoint which might explain why the host didn't get the response of the first setup request.
If set to 0, hardware will try to transfer 0 bytes to the host which is obviously wrong.